packages/shared/src/hooks/utils/useDownloadUrl.ts

```diff
@@ -0,0 +1,35 @@
+interface DownloadProps {
+  filename: string;
+  url: string;
+}
+
+interface UseDownloadUrl {
+  onDownloadUrl: (props: DownloadProps) => Promise<void>;
+}
+
+interface UseDownloadUrlProps {
+  onError?: (error: Error) => void;
+}
+
+export const useDownloadUrl = ({
+  onError,
+}: UseDownloadUrlProps = {}): UseDownloadUrl => {
+  const onDownloadUrl = async ({ filename, url }: DownloadProps) => {
```

Comment:
memoize with useCallback

Reply:
This could also be react-query mutation where you would get loading, error and all other states for free which might be even better?

Reply:
Totally agree with that. Didnt realize it at first as i just copy pasted this part.